### PR TITLE
FOGL-7138 Fix for missing configuration item properties

### PR DESCRIPTION
**Pull request by MarkRiddoch**
Signed-off-by: Mark Riddoch <mark@dianomic.com>

### CODE REVIEW

File: C/common/include/config_category.h [hunk 1]
```diff
@@ -143,7 +145,7 @@ class ConfigCategory {
 					     const std::vector<std::string> options);
 				CategoryItem(const CategoryItem& rhs);
 				// Return both "value" and "default" items
-				std::string	toJSON(const bool full=false) const;
+				std::string	toJSON(const bool full=true) const;
```
Comment (pintomax) [inline C/common/include/config_category.h:148]: Why the optional parameter has a new default value? 

I do not recall exactly but some routines are calling  toJSON() and others toJSON(true)

**Review by pintomax — CHANGES_REQUESTED** [review]

**Review by ashish-jabble — CHANGES_REQUESTED** [review]
@MarkRiddoch C based unit tests are failing. So I am not sure if it's worth to look upon the testing for the missing attributes when merge configuration called from C North sending process.
Though I have tried to test
A) OMF north as a service with default configuration, see fatal errors in syslogs though service is running fine.
```
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: Signal 6 (Aborted) trapped:
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (0) 0       0x55d465367dc9 handler(int) + 73---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (1) 1       0x7f13a252ef10 /lib/x86_64-linux-gnu/libc.so.6(+0x3ef10) [0x7f13a252ef10]---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (2) 2       0x7f13a252ee87 gsignal + 199---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (3) 3       0x7f13a25307f1 abort + 321---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (4) 4       0x7f13a2b85957 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8c957) [0x7f13a2b85957]---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (5) 5       0x7f13a2b8bae6 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92ae6) [0x7f13a2b8bae6]---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (6) 6       0x7f13a2b8bb21 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92b21) [0x7f13a2b8bb21]---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (7) 7       0x7f13a2b8bd54 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92d54) [0x7f13a2b8bd54]---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (8) 8       0x7f13a3377b61 ConfigCategory::ConfigCategory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 1441---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (9) 9       0x55d465369332 NorthService::configChange(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 162---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (10) 10      0x7f13a302c948 ConfigHandler::configChange(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 300---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (11) 11      0x7f13a3032551 ManagementApi::configChange(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>) + 337---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (12) 12      0x7f13a3031479 configChangeWrapper(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>) + 102---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (13) 13      0x7f13a304f4d7 std::_Function_handler<void (std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>), void (*)(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>)>::_M_invoke(std::_Any_data const&, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>&&, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_soc
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (14) 14      0x7f13a305297f std::function<void (std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>)>::operator()(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>) const + 97---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (15) 15      0x7f13a304cf85 SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::write(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&, std::function<void (std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>)>&) + 211---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (16) 16      0x7f13a3047093 SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::find_resource(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&) + 1009---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (17) 17      0x7f13a3042e81 SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::read(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&)::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const + 161---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (18) 18      0x7f13a304d365 boost::asio::detail::read_streambuf_op<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >, std::allocator<char>, boost::asio::detail::transfer_exactly_t, SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::read(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&)::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const::{lambda(boost::system::error_code const&, unsigned long)#1}>::operator()(boost::system::error_code const&, unsigned long, int) + 371---------
Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (19) 19      0x7f13a3079aa0 boost::asio::detail::binder2<boost::asio::detail::read_streambuf_op<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >, std::allocator<char>, boost::asio::detail::transfer_exactly_t, SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::read(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&)::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const::{lambda(boost::system::error_code const&, unsigned long)#1}>, boost::system::error_code, unsigned long>::operator()() + 48---------
```
B) OMF north as a task with default configuration, also see some Fatal errors
```
Dec 22 11:23:03 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:23:05.721552
Dec 22 11:23:03 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Enabled Schedule 'OMF-P/954dbd72-220f-47f9-a499-5d3a156a7a5a' process 'north_c'
Dec 22 11:23:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'OMF-P' process 'north_c' task e07bca89-47ef-41ca-b343-7189c0b641ef pid 14167, 1 running tasks#012['tasks/north_c', '--port=45149', '--address=127.0.0.1', '--name=OMF-P']
Dec 22 11:23:06 aj-ThinkPad-E450 Fledge OMF-P[14168]: ERROR: Failed to parse result of adding a category: 400: Expecting ',' delimiter: line 1 column 1095 (char 1094)
Dec 22 11:23:06 aj-ThinkPad-E450 Fledge OMF-P[14168]: FATAL: Failure creating/updating configuration key 'OMF-P'
Dec 22 11:23:06 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process terminated: Schedule 'OMF-P' process 'north_c' task e07bca89-47ef-41ca-b343-7189c0b641ef pid 14167 exit 2, 0 running tasks#012['tasks/north_c']
Dec 22 11:23:06 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:23:35.721552
Dec 22 11:23:35 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'OMF-P' process 'north_c' task e0d26aa4-10ab-4f0a-b131-17f5e5adf660 pid 14331, 1 running tasks#012['tasks/north_c', '--port=45149', '--address=127.0.0.1', '--name=OMF-P']
Dec 22 11:23:35 aj-ThinkPad-E450 Fledge OMF-P[14332]: ERROR: Failed to parse result of adding a category: 400: Expecting ',' delimiter: line 1 column 1095 (char 1094)
Dec 22 11:23:35 aj-ThinkPad-E450 Fledge OMF-P[14332]: FATAL: Failure creating/updating configuration key 'OMF-P'
Dec 22 11:23:35 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process terminated: Schedule 'OMF-P' process 'north_c' task e0d26aa4-10ab-4f0a-b131-17f5e5adf660 pid 14331 exit 2, 0 running tasks#012['tasks/north_c']
Dec 22 11:23:35 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:24:05.721552
Dec 22 11:24:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'OMF-P' process 'north_c' task c50311ce-19bf-472f-81c6-83f8e63d2151 pid 14351, 1 running tasks#012['tasks/north_c', '--port=45149', '--address=127.0.0.1', '--name=OMF-P']
Dec 22 11:24:05 aj-ThinkPad-E450 Fledge OMF-P[14352]: ERROR: Failed to parse result of adding a category: 400: Expecting ',' delimiter: line 1 column 1095 (char 1094)
Dec 22 11:24:05 aj-ThinkPad-E450 Fledge OMF-P[14352]: FATAL: Failure creating/updating configuration key 'OMF-P'
Dec 22 11:24:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process terminated: Schedule 'OMF-P' process 'north_c' task c50311ce-19bf-472f-81c6-83f8e63d2151 pid 14351 exit 2, 0 running tasks#012['tasks/north_c']
Dec 22 11:24:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:24:35.721552

```

File: C/common/include/config_category.h (hunk 1 above)
Comment (MarkRiddoch) [reply C/common/include/config_category.h:148]: It is possibly not needed. The problem is caused by incomplete configuration categories being sent to the configuration manager for merge, hence always sending everything. Will experiment further to see if this is needed.

**Comment by MarkRiddoch** [comment]
> @MarkRiddoch C based unit tests are failing. So I am not sure if it's worth to look upon the testing for the missing attributes when merge configuration called from C North sending process. Though I have tried to test A) OMF north as a service with default configuration, see fatal errors in syslogs though service is running fine.
> 
> ```
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: Signal 6 (Aborted) trapped:
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (0) 0       0x55d465367dc9 handler(int) + 73---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (1) 1       0x7f13a252ef10 /lib/x86_64-linux-gnu/libc.so.6(+0x3ef10) [0x7f13a252ef10]---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (2) 2       0x7f13a252ee87 gsignal + 199---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (3) 3       0x7f13a25307f1 abort + 321---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (4) 4       0x7f13a2b85957 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8c957) [0x7f13a2b85957]---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (5) 5       0x7f13a2b8bae6 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92ae6) [0x7f13a2b8bae6]---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (6) 6       0x7f13a2b8bb21 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92b21) [0x7f13a2b8bb21]---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (7) 7       0x7f13a2b8bd54 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92d54) [0x7f13a2b8bd54]---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (8) 8       0x7f13a3377b61 ConfigCategory::ConfigCategory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 1441---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (9) 9       0x55d465369332 NorthService::configChange(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 162---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (10) 10      0x7f13a302c948 ConfigHandler::configChange(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 300---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (11) 11      0x7f13a3032551 ManagementApi::configChange(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>) + 337---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (12) 12      0x7f13a3031479 configChangeWrapper(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>) + 102---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (13) 13      0x7f13a304f4d7 std::_Function_handler<void (std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>), void (*)(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>)>::_M_invoke(std::_Any_data const&, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>&&, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_soc
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (14) 14      0x7f13a305297f std::function<void (std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>)>::operator()(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>) const + 97---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (15) 15      0x7f13a304cf85 SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::write(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&, std::function<void (std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Response>, std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Request>)>&) + 211---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (16) 16      0x7f13a3047093 SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::find_resource(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&) + 1009---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (17) 17      0x7f13a3042e81 SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::read(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&)::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const + 161---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (18) 18      0x7f13a304d365 boost::asio::detail::read_streambuf_op<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >, std::allocator<char>, boost::asio::detail::transfer_exactly_t, SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::read(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&)::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const::{lambda(boost::system::error_code const&, unsigned long)#1}>::operator()(boost::system::error_code const&, unsigned long, int) + 371---------
> Dec 22 11:18:03 aj-ThinkPad-E450 Fledge OMF[13152]: FATAL: (19) 19      0x7f13a3079aa0 boost::asio::detail::binder2<boost::asio::detail::read_streambuf_op<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >, std::allocator<char>, boost::asio::detail::transfer_exactly_t, SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::read(std::shared_ptr<SimpleWeb::ServerBase<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> > >::Session> const&)::{lambda(boost::system::error_code const&, unsigned long)#1}::operator()(boost::system::error_code const&, unsigned long) const::{lambda(boost::system::error_code const&, unsigned long)#1}>, boost::system::error_code, unsigned long>::operator()() + 48---------
> ```
> 
> B) OMF north as a task with default configuration, also see some Fatal errors
> 
> ```
> Dec 22 11:23:03 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:23:05.721552
> Dec 22 11:23:03 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Enabled Schedule 'OMF-P/954dbd72-220f-47f9-a499-5d3a156a7a5a' process 'north_c'
> Dec 22 11:23:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'OMF-P' process 'north_c' task e07bca89-47ef-41ca-b343-7189c0b641ef pid 14167, 1 running tasks#012['tasks/north_c', '--port=45149', '--address=127.0.0.1', '--name=OMF-P']
> Dec 22 11:23:06 aj-ThinkPad-E450 Fledge OMF-P[14168]: ERROR: Failed to parse result of adding a category: 400: Expecting ',' delimiter: line 1 column 1095 (char 1094)
> Dec 22 11:23:06 aj-ThinkPad-E450 Fledge OMF-P[14168]: FATAL: Failure creating/updating configuration key 'OMF-P'
> Dec 22 11:23:06 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process terminated: Schedule 'OMF-P' process 'north_c' task e07bca89-47ef-41ca-b343-7189c0b641ef pid 14167 exit 2, 0 running tasks#012['tasks/north_c']
> Dec 22 11:23:06 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:23:35.721552
> Dec 22 11:23:35 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'OMF-P' process 'north_c' task e0d26aa4-10ab-4f0a-b131-17f5e5adf660 pid 14331, 1 running tasks#012['tasks/north_c', '--port=45149', '--address=127.0.0.1', '--name=OMF-P']
> Dec 22 11:23:35 aj-ThinkPad-E450 Fledge OMF-P[14332]: ERROR: Failed to parse result of adding a category: 400: Expecting ',' delimiter: line 1 column 1095 (char 1094)
> Dec 22 11:23:35 aj-ThinkPad-E450 Fledge OMF-P[14332]: FATAL: Failure creating/updating configuration key 'OMF-P'
> Dec 22 11:23:35 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process terminated: Schedule 'OMF-P' process 'north_c' task e0d26aa4-10ab-4f0a-b131-17f5e5adf660 pid 14331 exit 2, 0 running tasks#012['tasks/north_c']
> Dec 22 11:23:35 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:24:05.721552
> Dec 22 11:24:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'OMF-P' process 'north_c' task c50311ce-19bf-472f-81c6-83f8e63d2151 pid 14351, 1 running tasks#012['tasks/north_c', '--port=45149', '--address=127.0.0.1', '--name=OMF-P']
> Dec 22 11:24:05 aj-ThinkPad-E450 Fledge OMF-P[14352]: ERROR: Failed to parse result of adding a category: 400: Expecting ',' delimiter: line 1 column 1095 (char 1094)
> Dec 22 11:24:05 aj-ThinkPad-E450 Fledge OMF-P[14352]: FATAL: Failure creating/updating configuration key 'OMF-P'
> Dec 22 11:24:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process terminated: Schedule 'OMF-P' process 'north_c' task c50311ce-19bf-472f-81c6-83f8e63d2151 pid 14351 exit 2, 0 running tasks#012['tasks/north_c']
> Dec 22 11:24:05 aj-ThinkPad-E450 Fledge[11580] INFO: scheduler: fledge.services.core.scheduler.scheduler: Scheduled task for schedule 'OMF-P' to start at 2022-12-22 11:24:35.721552
> ```

This is caused by quote in the validity and a missing escape call. This also fixes the existing categories.

**Comment by ashish-jabble** [comment]
@MarkRiddoch now looks good optional attributes are not missing either with service or task for newly instances. Also seems like `rule, deprecated, displayName` optional attributes handling is missing?

File: C/common/include/config_category.h [hunk 2]
```diff
@@ -124,7 +124,9 @@ class ConfigCategory {
 					FILE_ATTR,
 					MINIMUM_ATTR,
 					MAXIMUM_ATTR,
-					LENGTH_ATTR};
+					LENGTH_ATTR,
+					VALIDITY_ATTR,
+					GROUP_ATTR};
```
[the remark this answers is not in the document]
Comment (MarkRiddoch) [reply C/common/include/config_category.h:129]: They can be added but nothing uses them on the C++ side currently. I think I would rather do that under a separate Jira that is not blocking the release.

**Review by ashish-jabble — APPROVED** [review]

File: C/common/include/config_category.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashish-jabble) [reply C/common/include/config_category.h:129]: Sure, Let's create a JIRA to handle it separately.

**Pull request merged**